File: Shared/Framework/Models/Account/AuthenticationResponse.cs
namespace Framework.Models.Account
{
    public class AuthenticationResponse<TPerson, TNotification>
    {
        public bool Succeeded { get; set; }
        public bool IsLockedOut { get; set; }
        public bool IsNotAllowed { get; set; }
        public bool RequiresTwoFactor { get; set; }
        public bool UserProfileCompleted { get; set; } = false;
        public string? Token { get; set; }
        public int ExpiresIn { get; set; }
        public string RefreshToken { get; set; } = null!;
        public string? Message { get; set; }
        public IList<string>? Roles { get; set; }
        public TPerson? Person { get; set; }
        public TNotification[]? Notifications { get; set; }
    }
}

File: AdventureWorksLT2019/ReactTSApp/src/shared/apis/AuthenticationResponse.ts
// NTierOnTime.WebApiMessage.AuthenticationResponse
export interface AuthenticationResponse {
    succeeded: boolean;
    isLockedOut: boolean;
    isNotAllowed: boolean;
    requiresTwoFactor: boolean;
    token: string | null;
    expiresIn: number;
    refreshToken: string | null;
    entityID: number | null;
    roles: string[];
}
